### PR TITLE
CLOUDP-280230: Replace launcher with contract helper

### PR DESCRIPTION
**Pull request by josvazg**
Simplify contract testing:

- No `TestMain` preparations required.
- No need for external YAML files for infrastructure.
- No need for a AKO helm version, latest will be used by default.
- No explicit skipping and cleaning, it happens within the contract test run wrapper.
- No hard dependency of the helper in `kind`, assumes a working k8s config is available, which CI and make provide easily.

Note: Using `make` locally, the contract tests launch `kind` with them. On the `CI`, this is not needed because the workflow launched `kind`already before running the tests, so we use an existing env var in the launch script to avoid launching `kind` again.

### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: test/helper/contract/k8s.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+func k8sRecreate(ctx context.Context, k8sClient client.Client, obj client.Object) error {
+	err := k8sClient.Create(ctx, obj)
+	if err == nil {
+		return nil
+	}
+	if !k8sErrors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create object: %w", err)
+	}
+	if err := k8sClient.Update(ctx, obj); err != nil {
```
Comment (s-urbaniak) [inline test/helper/contract/k8s.go:83]: this is missing conflict resolution.

**Review by s-urbaniak — CHANGES_REQUESTED** [review]
one comment, else lgtm 👍

**Comment by josvazg** [comment]
CLOUDP-280230

File: test/helper/contract/k8s.go (hunk 1 above)
Comment (josvazg) [reply test/helper/contract/k8s.go]: Have added a retry on update.

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**